packages/@o3r-training/tools/src/utility/web-container.ts

```diff
@@ -0,0 +1,85 @@
+import type { BufferEncoding, DirEnt, FileSystemTree } from '@webcontainer/api';
+
+/**
+ * Function to read a directory in the file system
+ */
+export type ReadDirFn = (root: string, options: {
+  encoding?: BufferEncoding | null | undefined;
+  recursive?: boolean | undefined;
+  withFileTypes?: boolean;
+}) => Promise<DirEnt<string>[]>;
+
+/**
+ * Function to read a file in the file system
+ */
+export type ReadFileFn = (root: string, encoding?: BufferEncoding | null | undefined) => Promise<string>;
+
+/**
+ * File system operations
+ */
+export type FileSystem = {
+  /** Reads a given directory and returns its files and directories */
+  readdir: ReadDirFn;
+  /** Reads a given file */
+  readFile: ReadFileFn;
+};
+
+/**
+ * Reads a given directory and returns the files and their corresponding contents
+ * @param entry
+ * @param path
+ * @param fileSystem
+ * @param exclusionList
+ */
+const readDirRec = async (entry: DirEnt<string>, path: string, fileSystem: FileSystem, exclusionList: string[] = []):
+  Promise<FileSystemTree | undefined> => {
+  const entryPath = path + '/' + entry.name;
+  if (exclusionList.includes(entry.name)) {
+    return;
+  }
```

Comment:
```suggestion
  if (exclusionList.includes(entry.name)) {
    return;
  }
  const entryPath = path + '/' + entry.name;
```

(to reduce easily a bit of calculation)